sql/rowexec/dml.go

```diff
@@ -54,6 +54,13 @@ func (b *BaseBuilder) buildInsertInto(ctx *sql.Context, ii *plan.InsertInto, row
 		return nil, err
 	}
 
+	var insertTuples [][]sql.Expression
```

Comment:
This is going to be nil in some cases, which will mean the INSERT statement will panic. Consider this insert:

`insert into mytable (colA) select colB from a limit 1;`

You won't have a *plan.Values there.

A better option here is just to use the row to be inserted, like the existing code path does:

```go
func (i *insertIter) updateLastInsertId(ctx *sql.Context, row sql.Row) {
	if i.lastInsertIdUpdated {
		return
	}

	autoIncVal := i.getAutoIncVal(row)

	if i.hasAutoAutoIncValue {
		ctx.SetLastQueryInfo(sql.LastInsertId, autoIncVal)
		i.lastInsertIdUpdated = true
	}
}
```

So you just need a corresponding method to get the auto-uuid value from the row provided. Any reason that wouldn't work?

Reply:
Good point about other insert value sources. I noticed how `expression.AutoIncrement` is used as a marker in the row that the insert code uses to record the auto_increment value for last_insert_id. I thought I could avoid needing that type, but you're right that it'll be needed for other non-`values` clauses, plus it'll help keep the tuple checking code separate from this code.